mex/extractors/pipeline/base.py

```diff
@@ -37,93 +36,69 @@ def run_job_in_process(group_name: str = "default") -> "ExecutionResult":
+        (
+            run.end_time  # given in unix time notation
+            for run in publisher_runs
+            if run.end_time
+        ),
+        default=0.0,  # unix time notation for 1970
+    )
+
+    extractor_runs = instance.get_run_records(
+        filters=RunsFilter(
+            statuses=[DagsterRunStatus.SUCCESS],
+            tags={"status": "extract"},
         )
-        if jobs_are_running:
-            return SkipReason(
-                "No publishing because other jobs are running at the moment."
-            )
-
-        for group in extractor_group_names:
-            runs = instance.get_runs(
-                filters=RunsFilter(
-                    job_name=group,
-                    statuses=[
-                        DagsterRunStatus.SUCCESS,
-                        DagsterRunStatus.FAILURE,
-                        DagsterRunStatus.CANCELED,
-                    ],
-                )
-            )
-
-            recent_run = max(
-                (
-                    run
-                    for run in runs
-                    if ".dagster/scheduled_execution_time" in run.tags
-                    and run.tags[".dagster/scheduled_execution_time"]
-                    > last_run_time_publisher
-                ),
-                key=lambda run: run.tags[".dagster/scheduled_execution_time"],
-                default=None,
-            )
-
-            # Update completed flag based on whether a recent run was found
-            if recent_run is None or recent_run.status not in [
-                DagsterRunStatus.SUCCESS,
-                DagsterRunStatus.FAILURE,
-                DagsterRunStatus.CANCELED,
-            ]:
-                completed = False
-                return SkipReason(f"No complete run for job group '{group}' yet.")
-
-            # Update the latest extractor start time if the current run is newer
-            if (
-                recent_run
-                and recent_run.tags[".dagster/scheduled_execution_time"]
-                > latest_start_time_extractors
-            ):
-                latest_start_time_extractors = recent_run.tags[
-                    ".dagster/scheduled_execution_time"
-                ]
-
-        if completed:
-            # Update the cursor to the latest extractor start time
-            context.update_cursor(latest_start_time_extractors)
-            return RunRequest(
-                run_key=latest_start_time_extractors,
-                run_config={},
-            )
-
-        return SkipReason("No publisher run for other reasons.")
-
-    return monitor_jobs_sensor
+    )
+
+    newest_unpublished_extractor_run_ts = max(
+        (
+            run.end_time
+            for run in extractor_runs
+            if run.end_time and run.end_time > newest_publisher_run_ts
+        ),
+        default=None,
+    )
+
+    if newest_unpublished_extractor_run_ts is None:
+        return SkipReason("No complete unpublished run for any extractor job yet.")
```

Comment:
I find this a bit easier to understand because both comprehensions do exactly the same thing: find the newest run.

```suggestion
    newest_extractor_run_ts = max(
        (
            run.end_time
            for run in extractor_runs
            if run.end_time
        ),
        default=0.0,
    )

    if newest_publisher_run_ts > newest_extractor_run_ts:
        return SkipReason("No complete unpublished run for any extractor job yet.")
```